lib/ddtrace/contrib/sidekiq/patcher.rb

```diff
@@ -34,8 +34,38 @@ def patch
             config.server_middleware do |chain|
               chain.add(Sidekiq::ServerTracer)
             end
+
+            patch_server_internals
           end
         end
+
+        def patch_server_internals
+          # Sidekiq < 5.2.4 (before mperham/sidekiq@ddb0c8b3) doesn't require this until
+          # too late in the process (after `CLI#run` has been called)
+          require 'sidekiq/launcher'
```

Comment:
We normally avoid requiring external gem code in `ddtrace`, as to not affect gem initialization order in the host application. Some gems are picky with initialization order.

Also, if I'm reading this correctly, 'sidekiq/launcher' is only needed when running Sidekiq jobs, but not when simply enqueuing them. This means that many applications that only enqueue jobs would actually never need to `require 'sidekiq/launcher'`.

With that in mind, what do you think of removing the `require` here and instead making 
```ruby
          patch_server_heartbeat
          patch_server_job_fetch
          patch_server_scheduled_push
```
conditional on having `sidekiq/launcher` loaded? It seems to me that this would work for users of Sidekiq >= 5.2.4, which is over 2 years old, but not force us to explicitly load it for all users of `ddtrace`.

Reply:
I'm more than happy to only do this for Sidekiq >= 5.2.4, but just to be clear, this won't attempt to load `sidekiq/launcher` for all users of dd-trace, only those who call `use :sidekiq` in their configuration. Does that change your opinion at all @marcotc?

Reply:
> only those who call use :sidekiq in their configuration

For web applications, for example, they will have `use :sidekiq` because they want to capture `MyJob.perform_async` calls. But these applications will never require `sidekiq/launcher` normally. Does that make sense?

Reply:
Sure, but we were only ever requiring `sidekiq/launcher` inside a `Sidekiq.configure_server` block, so it *shouldn't* have had a significant impact. Either way though, I've removed it!